pkg/generator/templates.go

```diff
@@ -130,37 +130,57 @@ spec:
 const mainTmpl = `package main
 
 import (
-	"context"
-	"runtime"
+  "context"
+  "runtime"
+  "net/http"
 
-	stub "{{.StubImport}}"
-	sdk "{{.OperatorSDKImport}}"
-	k8sutil "{{.K8sutilImport}}"
-	sdkVersion "{{.SDKVersionImport}}"
+  stub "{{.StubImport}}"
+  sdk "{{.OperatorSDKImport}}"
+  k8sutil "{{.K8sutilImport}}"
+  sdkVersion "{{.SDKVersionImport}}"
 
-	"github.com/sirupsen/logrus"
+  "github.com/prometheus/client_golang/prometheus/promhttp"
+  "github.com/sirupsen/logrus"
 )
 
 func printVersion() {
-	logrus.Infof("Go Version: %s", runtime.Version())
-	logrus.Infof("Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH)
-	logrus.Infof("operator-sdk Version: %v", sdkVersion.Version)
+  logrus.Infof("Go Version: %s", runtime.Version())
+  logrus.Infof("Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH)
+  logrus.Infof("operator-sdk Version: %v", sdkVersion.Version)
+  logrus.Infof("operator prometheus port %s", "{{.MetricsPort}}")
 }
 
-func main() {
-	printVersion()
+func initOperatorService() {
+  service, err := k8sutil.InitOperatorService()
+  if err != nil {
+    logrus.Fatalf("Failed to init operator service: %v", err)
+  }
+  err = sdk.Create(service)
+  if err != nil {
+    logrus.Infof("Failed to create operator service: %v", err)
```

Comment:
We should ignore the error if the service already exists as it's very likely that the operator pod can restart.
```Go
import "k8s.io/apimachinery/pkg/api/errors"
...
if err != nil && !errors.IsAlreadyExists(err) {
    logrus.Infof("Failed to create operator service: %v", err)
}
```